e2e/pages/page.ts

```diff
@@ -69,19 +54,15 @@ export abstract class Page {
   }
 
   async waitForApp() {
-    await browser.wait(EC.presenceOf(this.layout), BROWSER_WAIT_TIMEOUT);
-  }
-
-  async waitForSnackBarToAppear() {
-    return browser.wait(until.elementLocated(by.css('.mat-snack-bar-container')), BROWSER_WAIT_TIMEOUT, '------- timeout waiting for snackbar to appear');
+    await waitForPresence(this.layout);
   }
 
   async waitForSnackBarToClose() {
-    await browser.wait(EC.not(EC.visibilityOf(this.snackBarContainer)), BROWSER_WAIT_TIMEOUT);
+    await waitForVisibility(this.snackBarContainer);
```

Comment:
here the call is for waitForVisibility, initially it was waiting for NOT visible

based on experience, I suggest to leave this as **EC.not(EC.visibilityOf... )** and to NOT replace this with waitForInvisibility.

Due to a protractor weirdness, in specific situations - like when dealing with elements that appear and then disappear, EC.not(EC.visibilityOf..) is not equivalent with EC.invisibilityOf(..).

Reply:
I am going to use invisitibityOf because it's exactly what protractor does inside (not + visibility) see here: https://github.com/angular/protractor/blob/5d8da049815ffc104e6eca71d2ecaec51b00f520/lib/expectedConditions.ts#L412